sdk/js/src/DefaultCacheStore.ts

```diff
@@ -1,26 +1,39 @@
-export class DefaultCacheStore {
-    private store?: Storage
+// Abstract Storage class
+import { DVCStorage } from './types'
+import { checkIsServiceWorker } from './utils'
 
-    constructor() {
-        this.store =
-            typeof window !== 'undefined'
-                ? window.localStorage
-                : stubbedLocalStorage
-    }
+export abstract class StorageStrategy implements DVCStorage {
+    abstract store: unknown
+    abstract init(): Promise<unknown>
+    abstract save(storeKey: string, data: unknown): Promise<void>
+    abstract load<T>(storeKey: string): Promise<T | undefined>
+    abstract remove(storeKey: string): Promise<void>
+}
+
+// LocalStorage implementation
+export class LocalStorageStrategy extends StorageStrategy {
+    override store: Storage
+    private isTesting: boolean
 
-    save(storeKey: string, data: unknown): void {
-        this.store?.setItem(storeKey, JSON.stringify(data))
+    constructor(isTesting = false) {
+        super()
+        this.isTesting = isTesting
+        this.init()
+    }
+    async init(): Promise<void> {
```

Comment:
is `init()` ever called out-side of the constructor? could this just be in the constructor?

Reply:
its not, I mostly did this because having an `init` function is more readable for the `IndexedDBStrategy` implementation and I wanted both classes to have the same interface. I can change this though